papers/views.py

```diff
@@ -124,6 +124,28 @@ def get(self, request, *args, **kwargs):
             request.GET.pop('visible', None)
             request.GET.pop('availability', None)
             request.GET.pop('oa_status', None)
+
+        # Handle search by ORCID
+        query = self.get_form_kwargs()['data']
+        if 'authors' in query:
+            orcids = [
+                item
+                for item in query['authors'].split()
+                if validate_orcid(item)
+            ]
+            for orcid in orcids:
+                try:
+                    researcher = Researcher.objects.get(orcid=orcid)
+                    researchers.append(researcher.id)
+                except MetadaSourceException:
+                    pass
+            authors = [
+                item
+                for item in query['authors'].split()
+                if item not in orcids
+            ]
+            if researchers:
+                self.queryset = self.queryset.filter(researchers=researchers)
```

Comment:
I am not sure about the syntax for SearchQuerySets but for Django's standard query sets, filtering on a property with a list of possible values, you need to use `filter(researchers__in=researchers)`.

Reply:
I probably have to do the same indeed, thanks for the tip! Even if `researchers` is not a list however, it does not work. I probably have a logic error here, I'll check this.